rfcs/0028-nix-release-model.md

```diff
@@ -0,0 +1,158 @@
+   docs, etc.) should target the `#-stabilization` branch and then
+   be merged forward into `master`. No commit should exist on the
+   `#-stabilization` branch without very soon thereafter being merged
+   into `master` except under rare circumstances. This implies that
+   especially large changes ready around a release should either be
+   included before stabilization branch-off (with extra time to
+   stabilize the new feature) or should, if possible, wait to be
+   merged to `master` until after the release. Which path is taken
+   should be a collaboration between the release manager and the
+   developers of the change in question.
+10. New development unrelated to the new release can go directly into
+    `master` in parallel with a release stabilization.
+11. During stabilization, the release manager should update
+    `nixUnstable` to point to various commits of the `#-stabilization`
+    branch, to easily enable wider testing and integration.
+12. When the release manager determines readiness, the relevant commit
+    is announced to the community. Barring objections or last-minute
+    fixes judged valid by the release manager (or core team), the
+    commit is tagged and the branch receives no further commits.
+    Normal development against `master` continues. Readiness is a
+    judgment call, and should require increased scrutiny/validation
+    for a release with more complex/major changes.
+13. Associated with each `NixOS` stable release is an associated
+    `maintained` series of `Nix`, with the same support lifetime as
+    the stable release.
+14. If `master` has not yet diverged enough to warrant a minor semver
+    bump relative to the latest release of a `maintained` series, a
+    maintenance release can follow the normal stabilization workflow.
+15. If `master` has diverged too much, a `#-maintenance` branch (e.g.
+    `2.1-maintenance` can be created.
+16. Any important bugfix can be cherry-picked from `master` to a
+    `#-maintenance` branch at any committer's discretion.
+17. In rare cases, if a change truly only makes sense on a
+    `#-maintenance` branch (e.g. when we added support to `1.11` to
+    be able to work with a `2.0` database schema) it can be targeted
+    directly to the `#-maintenance` branch. This should be avoided if
+    possible and subject to especially careful testing and review.
+18. A maintenance release can be tagged off of the `#-maintenance`
+    branch at any time *after* there is a mainline release that
+    contains all of the fixes included in the desired maintenance
+    release, at the discretion of any Nix core team member.
+19. \*Automated testing, ideally applied to all mainline PRs, should
+     ensure that all releases of all supported maintained series are
+     able to successfully install master if a release were cut. This
+     should include a number of definitions of "install", especially
+     `NixOS` upgrades.
```

Comment:
The list could be split up to only include the release process

Reply:
Not sure what you mean?

Reply:
Right now the list contains general ideas like keeping master stable that don't need to be numbered. It makes the list more impressive than it has to be.